src/vsphere_cpi/lib/cloud/vsphere/nsxt_provider.rb

```diff
@@ -393,12 +395,21 @@ def to_simple_expressions(lports)
     end
 
     def nsxt_nics(vm)
-      nics = vm.nics.select do |nic|
-        nic.backing.is_a?(VimSdk::Vim::Vm::Device::VirtualEthernetCard::OpaqueNetworkBackingInfo) &&
-          nic.backing.opaque_network_type == NSXT_LOGICAL_SWITCH
+      nics = []
+      vm.nics.select do |nic|
```

Comment:
I think this should be `vm.nics.each` rather than `vm.nics.select`. Overall I feel like this function is a little complicated. Here's my take:

```ruby
nics = vm.nics.select do |nic|
  case nic.backing
  when VimSdk::Vim::Vm::Device::VirtualEthernetCard::OpaqueNetworkBackingInfo
    next unless nic.backing.opaque_network_type == NSXT_LOGICAL_SWITCH
    logger.info("NIC with device key #{nic.key} is backed by opaque NSXT network")
  when VimSdk::Vim::Vm::Device::VirtualEthernetCard::DistributedVirtualPortBackingInfo
    next unless @vcenter_client.dvpg_istype_nsxt?(
      key: nic.backing.port.portgroup_key, dc: @datacenter
    )
    logger.info("NIC with device key #{nic.key} is backed by NSXT DVPG")
  end
  true
end
logger.info("NSX-T networks found for vm '#{vm.cid}': #{nics.map(&:device_info).map(&:summary)}")
nics
```